sky/unit/test/widget/input_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:sky/services.dart';
+import 'package:sky/widgets.dart';
+import 'package:sky/src/services/shell.dart' as shell;
+import 'package:test/test.dart';
+
+import 'widget_tester.dart';
+
+class MockKeyboard implements KeyboardService {
+  KeyboardClient client;
+
+  void show(Object client, int type) {
+    this.client = client.impl;
+  }
+
+  void showByRequest() {}
+
+  void hide() {}
+}
+
+void main() {
+  test('Editable text has consistent width', () {
+    WidgetTester tester = new WidgetTester();
+
+    MockKeyboard mockKeyboard = new MockKeyboard();
+    shell.overrideRequestService = (url) {
+      if (url == 'mojo:keyboard') {
+        return mockKeyboard;
+      }
+    };
```

Comment:
This isn't quite right.  The `mojo:keyboard` app can offer any number of services.  You need to look at the runtimeType of the proxy to see if its a KeyboardProxy and only then override it with the mock keyboard.